src/socket.c

```diff
@@ -1041,6 +1041,56 @@ void pthreads_socket_get_last_error(zval *object, zend_bool clear, zval *return_
 	}
 }
 
+void pthreads_socket_create_listen(zend_long port, zend_long backlog, zval *return_value) {
+	struct sockaddr_in  la;
+	struct hostent		*hp;
+	pthreads_object_t	*created;
+	pthreads_socket_t	*sock;
+
+	object_init_ex(return_value, pthreads_socket_entry);
+
+	created = PTHREADS_FETCH_FROM(Z_OBJ_P(return_value));
+	sock = created->store.sock;
+
+#ifndef PHP_WIN32
+	if ((hp = php_network_gethostbyname("0.0.0.0")) == NULL) {
+#else
+	if ((hp = php_network_gethostbyname("localhost")) == NULL) {
+#endif
+		zval_ptr_dtor(return_value);
```

Comment:
@tpunt is `zval_ptr_dtor` used correctly at this point and below?

Reply:
This looks fine to me. Technically, `zval_dtor` could be used, since we know that the newly created socket object will not contain a cycle (so there's no need to place it into the GC root buffer). But it doesn't really matter.